app/src/main/java/org/p2p/wallet/home/ui/wallet/WalletPresenter.kt

```diff
@@ -109,13 +166,67 @@ class WalletPresenter(
     }
 
     override fun onTopupClicked() {
-        // TODO
+        view?.showTopupWalletDialog()
     }
 
     override fun onSendClicked(clickSource: SearchOpenedFromScreen) {
         // TODO
     }
 
+    override fun onStrigaOnRampClicked(item: StrigaOnRampCellModel) {
+        launch {
+            try {
+                view?.showStrigaOnRampProgress(isLoading = true, tokenMint = item.tokenMintAddress)
+                val challengeId = strigaOnRampInteractor.onRampToken(item.amountAvailable, item.payload).unwrap()
+                view?.navigateToStrigaOnRampOtp(
+                    item.amountAvailable.asUsd(),
+                    challengeId
+                )
+            } catch (e: Throwable) {
+                Timber.e(e, "Error on on-ramping striga token")
+                if (BuildConfig.DEBUG) {
+                    view?.showErrorMessage(IllegalStateException("Striga Claim is not supported yet", e))
+                } else {
+                    view?.showUiKitSnackBar(messageResId = R.string.error_general_message)
+                }
+            } finally {
+                view?.showStrigaOnRampProgress(isLoading = false, tokenMint = item.tokenMintAddress)
+            }
+        }
+    }
+
+    override fun onStrigaBannerClicked(item: StrigaBanner) {
+        with(item.status) {
+            val statusFromKycBanner = homeMapper.getKycStatusBannerFromTitle(bannerTitleResId)
+            when {
+                statusFromKycBanner == StrigaKycStatusBanner.PENDING -> {
+                    view?.showKycPendingDialog()
+                }
+                statusFromKycBanner != null -> {
+                    launch {
+                        // hide banner if necessary
+                        homeInteractor.hideStrigaUserStatusBanner(statusFromKycBanner)
+
+                        if (statusFromKycBanner == StrigaKycStatusBanner.VERIFICATION_DONE) {
+                            view?.showStrigaBannerProgress(isLoading = true)
+
+                            homeInteractor.loadDetailsForStrigaAccounts()
+                                .onSuccess { view?.navigateToStrigaByBanner(statusFromKycBanner) }
+                                .onFailure { view?.showUiKitSnackBar(messageResId = R.string.error_general_message) }
+
+                            view?.showStrigaBannerProgress(isLoading = false)
+                        } else {
+                            view?.navigateToStrigaByBanner(statusFromKycBanner)
+                        }
```

Comment:
```suggestion
                        if (statusFromKycBanner == StrigaKycStatusBanner.VERIFICATION_DONE) {
                            view?.showStrigaBannerProgress(isLoading = true)

                            homeInteractor.loadDetailsForStrigaAccounts()
                                .onSuccess { view?.navigateToStrigaByBanner(statusFromKycBanner) }
                                .onFailure { view?.showUiKitSnackBar(messageResId = R.string.error_general_message) }

                            view?.showStrigaBannerProgress(isLoading = false)
                        } else {
                            view?.navigateToStrigaByBanner(statusFromKycBanner)
                        }
```

let's extract this branch to a method

Reply:
done; extracted to a handler